dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/FfdcCleanupTest.java

```diff
@@ -0,0 +1,142 @@
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+
+import componenttest.annotation.ExpectedFFDC;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.impl.LibertyServerFactory;
+import componenttest.topology.utils.HttpUtils;
+
+/**
+ *
+ */
+@RunWith(FATRunner.class)
+public class FfdcCleanupTest {
+    private static final String SERVER_NAME_XML = "com.ibm.ws.logging.ffdcCleanup";
+    private static final int CONN_TIMEOUT = 10;
+    private static LibertyServer server;
+
+    @BeforeClass
+    public static void initialSetup() throws Exception {
+        server = LibertyServerFactory.getLibertyServer(SERVER_NAME_XML);
+        server.saveServerConfiguration();
+        ShrinkHelper.defaultDropinApp(server, "ffdc-servlet", "com.ibm.ws.logging.fat.ffdc.servlet");
+
+    }
+
+    public void setUp() throws Exception {
+        if (server != null && !server.isStarted()) {
+            // Restore the original server configuration, before starting the server for each test case.
+            server.restoreServerConfiguration();
+            server.startServer();
+        }
+    }
+
+    @After
+    public void cleanUp() throws Exception {
+        if (server != null && server.isStarted()) {
+            server.stopServer("com.ibm.ws.logging.fat.ffdc.servlet", "ArithmeticException", "SRVE0777E");
+        }
+    }
+
+    @Test
+    @ExpectedFFDC("java.lang.ArithmeticException")
+    public void testFfdcCleanupDeletion() throws Exception {
+        setUp();
+        hitWebPage("ffdc-servlet", "FFDCServlet", true, "?generateFFDC=true");
+
+        ServerConfiguration serverConfig = server.getServerConfiguration();
+        Logging loggingObj = serverConfig.getLogging();
+        Thread.sleep(61000);
+        loggingObj.setMaxFfdcAge("0m");
+        server.updateServerConfiguration(serverConfig);
+
+        RemoteFile messagesLogFile = server.getDefaultLogFile();
+        String line = server.waitForStringInLogUsingMark("based on the maxFfdcAge value configured.", messagesLogFile);
+        assertTrue("The FFDC file was not deleted.", line.contains("Deleted 1 FFDC file"));
+    }
+
+    @Test
+    @ExpectedFFDC("java.lang.ArithmeticException")
+    public void testFfdcCleanupNoDeletion() {
+        try {
+            setUp();
+            hitWebPage("ffdc-servlet", "FFDCServlet", true, "?generateFFDC=true");
+
+            ServerConfiguration serverConfig = server.getServerConfiguration();
+            Logging loggingObj = serverConfig.getLogging();
+            Thread.sleep(61000);
+            loggingObj.setMaxFfdcAge("1d");
+            server.updateServerConfiguration(serverConfig);
+
+            RemoteFile messagesLogFile = server.getDefaultLogFile();
+            String line = server.waitForStringInLogUsingMark("based on the maxFfdcAge value configured.", messagesLogFile);
+            assertNull("The FFDC file was deleted.", line);
```

Comment:
It may be less confusing if the message is "The FFDC file was incorrectly deleted."